apps/app/src/views/components/TreatmentOptionSearch.tsx

```diff
@@ -0,0 +1,123 @@
+import { Box, ChakraProvider, InputGroup, InputRightElement, Spinner } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { gql, useApolloClient } from '@apollo/client';
+import { usePhoton } from '@photonhealth/react';
+import { useDebounce } from 'use-debounce';
+import { SelectField } from './SelectField';
+
+const SearchTreatmentOptionsQuery = gql`
+  query SearchTreatmentOptions($searchTerm: String!) {
+    treatmentOptions(searchTerm: $searchTerm) {
+      id: medicationId
+      form
+      name
+      ndc
+      route
+      strength
+      type
+      __typename
+    }
+  }
+`;
+
+type SelectedTreatment = {
+  medicineId: string;
+  name: string;
+};
+
+type TreatmentOptionSearchProps = {
+  setSelectedTreatment: (s: SelectedTreatment | undefined) => void;
+};
+
+export const TreatmentOptionSearch = ({
+  setSelectedTreatment: setSelectedTreatment
+}: TreatmentOptionSearchProps) => {
+  const client = usePhoton();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const [treatmentOptions, setTreatmentOptions] =
+    useState<Array<{ value: string; label: string }>>();
+
+  const clinicalClient = useApolloClient(client?.clinicalClient);
```

Comment:
is there a more elegant way to reference the correct apollo client? the default one points to the lambdas; the medisearch is on the clinical api.

Reply:
naw, this is current state of the art